File: LongLib/src/main/java/com/hloong/lib/longlog/LongLogManager.java
package com.hloong.lib.longlog;

import com.hloong.lib.longlog.base.LongLogConfig;
import com.hloong.lib.longlog.base.LongLogPrinter;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class LongLogManager {
    private LongLogConfig config;
    private static LongLogManager instance;
    private List<LongLogPrinter> printers = new ArrayList<>();
    private LongLogManager(LongLogConfig config,LongLogPrinter[] printers){
        this.config = config;
        this.printers.addAll(Arrays.asList(printers));

    }
    public static LongLogManager getInstance(){
        return instance;
    }
    public static void init(@NotNull LongLogConfig config,LongLogPrinter... printers){
        instance = new LongLogManager(config,printers);
    }

    public LongLogConfig getConfig(){
        return config;
    }

    public List<LongLogPrinter> getPrinters() {
        return printers;
    }

    public void addPrinters(LongLogPrinter printer) {
        printers.add(printer);
    }
    public void removePrinters(LongLogPrinter printer) {
        if (printer != null){
            printers.remove(printer);
        }
    }
}

File: LongLib/src/main/java/com/hloong/lib/longlog/LongLogManager.kt
package com.hloong.lib.longlog

import com.hloong.lib.longlog.base.LongLogConfig
import com.hloong.lib.longlog.base.LongLogPrinter
import java.util.*

class LongLogManager private constructor(var config: LongLogConfig, printers: Array<out LongLogPrinter>) {
    private var printers: MutableList<LongLogPrinter> = ArrayList()

    init {
        this.printers.addAll(listOf(*printers))
    }

    fun getPrinters(): List<LongLogPrinter> {
        return printers
    }

    fun addPrinters(printer: LongLogPrinter) {
        printers.add(printer)
    }

    fun removePrinters(printer: LongLogPrinter?) {
        if (printer != null) {
            printers.remove(printer)
        }
    }

    companion object {
        @JvmStatic
        var instance: LongLogManager? = null
            private set

        fun init(config: LongLogConfig, vararg printers: LongLogPrinter) {
            instance = LongLogManager(config, printers)
        }
    }
}